### PR TITLE
coqdep: add `-I` from `-f _CoqProject` to findlib search path, remove `-m`

### PR DESCRIPTION
**Pull request by SkySkimmer**
This should make it possible to implement coqdep's fl.ml using more standard findlib APIs in a further PR (possibly https://github.com/coq/coq/pull/19857)

### CODE REVIEW

File: test-suite/misc/non-marshalable-state/_CoqProject [hunk 1]
```diff
@@ -1,6 +1,7 @@
 META.coq-test-suite
 -Q theories Marshal
 -I src
+-I .
```
Comment (gares) [inline test-suite/misc/non-marshalable-state/_CoqProject:4]: Why the META file is still here, and not src/, and why it all works?

Comment (SkySkimmer) [reply test-suite/misc/non-marshalable-state/_CoqProject]: it works because this isn't testing coq_makefile

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ejgallego — APPROVED** [review]
Parts I'm familiar with look good to me.

File: tools/coqdep/lib/args.ml [hunk 2]
```diff
@@ -67,22 +89,19 @@ let parse st args =
...
-       parse { st with vo_path = (false, r, ln) :: st.vo_path } ll
-    | "-R" :: ([] | [_]) -> usage ()
+       parse (add_vo_path st (false, r, ln)) ll
+    | ("-Q"|"-R") :: ([] | [_]) -> usage ()
     | "-exclude-dir" :: r :: ll -> System.exclude_directory r; parse st ll
     | "-exclude-dir" :: [] -> usage ()
     | "-coqlib" :: r :: ll -> Boot.Env.set_coqlib r; parse st ll
     | "-coqlib" :: [] -> usage ()
     | "-dyndep" :: dyndep :: ll -> parse { st with dyndep } ll
-    | "-m" :: m :: ll -> parse { st with meta_files = st.meta_files @ [m]} ll
```
Comment (ejgallego) [inline tools/coqdep/lib/args.ml:85]: Maybe you want to emit a deprecation / warning message for users of `-m` ?

Comment (SkySkimmer) [reply tools/coqdep/lib/args.ml:85]: I don't think so

**Comment by SkySkimmer** [comment]
@coq/build-maintainers @coq/coq-makefile-maintainers can I get an assignee?

File: doc/changelog/09-cli-tools/19863-coqdep-nom.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- **Removed:**
+  `coqdep` flag `-m` (it was used through `coq_makefile`)
```
Comment (silene) [inline doc/changelog/09-cli-tools/19863-coqdep-nom.rst:2]: Actually, the flag was also used by people who manually call `coqdep`, e.g., to produce a graph of dependencies such as https://coqinterval.gitlabpages.inria.fr/interval/

Comment (SkySkimmer) [reply doc/changelog/09-cli-tools/19863-coqdep-nom.rst:2]: That should probably be using -I like the coqc calls.

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
@coqbot merge now

**Pull request merged**